retrieval/series_cache.go

```diff
@@ -196,7 +195,7 @@ func (c *seriesCache) garbageCollect() error {
 	if cpNum <= c.lastCheckpoint {
 		return nil
 	}
-	sr, err := wal.NewSegmentsReader(filepath.Join(c.dir, cpDir))
```

Comment:
why do we remove `c.dir` parent directory?

Reply:
Because the WAL API changed and `cpDir` now contains what before was `filepath.Join(c.dir, cpDir)`

Reply:
I learned that it was `tsdb.LastCheckpoint()` changed API (CHANGELOG 0.3.0 in [change log](https://github.com/prometheus/tsdb/blob/v0.10.0/CHANGELOG.md#030))